### PR TITLE
Vulkan rules relaxed for glslang C interface

### PR DESCRIPTION
**Pull request by chaoticbob**
Hi,

This PR adds Vulkan rules relaxed to the glslang C interface to make it more useful for Vulkan. I wasn't able to find a precedent for how this behavior should be added so I added a member variable to glslang_input_t. If this doesn't feel like a natural usage to the C interface I'm happy to make any adjustments. PTAL. 

Cheers,
Hai

### CODE REVIEW

**Review by jeremy-lunarg — APPROVED** [review]
LGTM

**Review by greg-lunarg — CHANGES_REQUESTED** [review]
While this would work, I think even better would be to add to the interface you created with your last PR (glslang_shader_set_options) and add this option along with the auto location and auto binding options.

The reason I would prefer that approach is that it would avoid changing the externally visible structure glslang_input_t which can "break" existing code using this interface.

Does this sound ok?

**Comment by chaoticbob** [comment]
@greg-lunarg Yes, that makes sense. Will make the appropriate changes.

**Comment by chaoticbob** [comment]
@greg-lunarg I made the changes per your suggestion. PTAL.

File: glslang/CInterface/glslang_c_interface.cpp [hunk 1]
```diff
@@ -367,6 +367,11 @@ GLSLANG_EXPORT void glslang_shader_set_options(glslang_shader_t* shader, int opt
     if (options & GLSLANG_SHADER_AUTO_MAP_LOCATIONS) {
         shader->shader->setAutoMapLocations(true);
     }
+
+    if (options & GLSLANG_SHADER_AUTO_MAP_LOCATIONS) {
```
Comment (greg-lunarg) [inline glslang/CInterface/glslang_c_interface.cpp:371]: Cut and paste error?

**Review by greg-lunarg — CHANGES_REQUESTED** [review]
I think just one problem, otherwise good.

File: glslang/CInterface/glslang_c_interface.cpp (hunk 1 above)
Comment (chaoticbob) [reply glslang/CInterface/glslang_c_interface.cpp:371]: Totally. This what I get for coding at 4AM. Will fix. Sorry.

**Comment by chaoticbob** [comment]
@greg-lunarg Fixed the copy/paste error.

**Review by greg-lunarg — APPROVED** [review]
LGTM. Thanks for the code!

**Pull request merged**